splight_lib/models/hub.py

```diff
@@ -166,9 +166,13 @@ def upload(cls, path: str):
         raw_hub_component = hub_client.get(
             name=name, version=version, first=True
         )
-        if not raw_hub_component:
-            raw_hub_component = HubComponent(**spec).save()
-        hub_component = HubComponent(**raw_hub_component)
+
+        new_hub_component = HubComponent(**spec)
+        if raw_hub_component:
+            old_hub_component = HubComponent(**raw_hub_component)
+            new_hub_component.id = old_hub_component.id
+        new_raw_hub_component = new_hub_component.save()
+        hub_component = HubComponent(**new_raw_hub_component)
```

Comment:
There are a few inconsistent steps here:
First, we should use `model_validate` method, for example, `HubComponent.model_validate(spec)`.
Also, I would do it as follow:
```python
hub_component = HubComponent.model_validate(spec)
if raw_hub_component:
    hub_component.id = raw_hub_component.get("id", None)
hub_component.save()
```

The `save()` method is in place, or it should it.

Reply:
noup `save()` it's not an inplace method also I think it's a good idea to validate the `raw_hub_component` and not just using  `raw_hub_component.get("id", None)` but I do agree I should use `model_validate`

Reply:
If `save()` is not inplace we have a bug there, it hsould be, as it is for the other models